[ISSUE] Panic when destroying a workspace with databricks_compliance_security_profile_workspace_setting

### Configuration
<!-- Please provide a minimal reproducible configuration for the issue -->
```hcl
resource "databricks_compliance_security_profile_workspace_setting" "compliance" {
  provider = databricks.workspace

  compliance_security_profile_workspace {
    is_enabled           = length(var.compliance_security_profile) > 0
    compliance_standards = var.compliance_security_profile
  }
}

```

### Expected Behavior

I should be able to destroy a workspace with a compliance security profile.

### Actual Behavior

│ Error: cannot delete compliance security profile workspace setting: panic: runtime error: invalid memory address or nil pointer dereference

### Steps to Reproduce

1. `terraform plan -destroy`
2. `terraform apply`

### Terraform and provider versions
Terraform v1.7.3

### Is it a regression?
<!-- Did this work in a previous version of the provider? If so, which versions did you try? -->

### Debug Output
<!-- Please add turn on logging, e.g. `TF_LOG=DEBUG terraform apply -no-color` and run command again, paste it to gist & provide the link to gist. If you're still willing to paste in log output, make sure you provide only relevant log lines with requests. It would make it more readable, if you pipe the log through `| grep databricks | sed -E 's/^.* plugin[^:]+: (.*)$/\1/'`, e.g.: `TF_LOG=DEBUG terraform apply -no-color 2>&1 | grep databricks | sed -E 's/^.* plugin[^:]+: (.*)$/\1/' 2>&1 |tee tf-debug.log`. If Terraform produced a panic, please provide a link to a GitHub Gist containing the output of the `crash.log`. -->

### Important Factoids
<!-- Are there anything atypical about your accounts that we should know? -->

### Would you like to implement a fix?
<!-- If you plan to implement a fix for this, let the maintainers and community know -->

@alexott : I observed similar error today:

databricks_compliance_security_profile_workspace_setting.this: Destroying... [id=global]
╷
│ Error: cannot delete compliance security profile workspace setting: panic: runtime error: invalid memory address or nil pointer dereference

Debug log:
2024-08-09T16:25:24.361-0400 [INFO]  provider.terraform-provider-databricks_v1.47.0: Explicit and implicit attributes: host: tf_provider_addr=registry.terraform.io/databricks/databricks tf_req_id=b92d9634-c20f-0202-4019-9ae3baeb80ad tf_rpc=Configure @caller=/home/runner/work/terraform-provider-databricks/terraform-provider-databricks/provider/provider.go:272 @module=databricks timestamp=2024-08-09T16:25:24.361-0400
2024-08-09T16:25:24.362-0400 [INFO]  Starting apply for databricks_compliance_security_profile_workspace_setting.this
2024-08-09T16:25:24.362-0400 [DEBUG] databricks_compliance_security_profile_workspace_setting.this: applying the planned Delete change
2024-08-09T16:25:24.363-0400 [ERROR] provider.terraform-provider-databricks_v1.47.0: Response contains error diagnostic: diagnostic_summary="cannot delete compliance security profile workspace setting: panic: runtime error: invalid memory address or nil pointer dereference" tf_proto_version=5.6 tf_provider_addr=registry.terraform.io/databricks/databricks @caller=/home/runner/work/terraform-provider-databricks/terraform-provider-databricks/vendor/github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/diag/diagnostics.go:58 diagnostic_detail= diagnostic_severity=ERROR tf_resource_type=databricks_compliance_security_profile_workspace_setting @module=sdk.proto tf_req_id=658b789d-4699-f271-d867-8c20b211cf38 tf_rpc=ApplyResourceChange timestamp=2024-08-09T16:25:24.363-0400
2024-08-09T16:25:24.370-0400 [DEBUG] State storage *statemgr.Filesystem declined to persist a state snapshot
2024-08-09T16:25:24.370-0400 [ERROR] vertex "databricks_compliance_security_profile_workspace_setting.this (destroy)" error: cannot delete compliance security profile workspace setting: panic: runtime error: invalid memory address or nil pointer dereference
2024-08-09T16:25:24.462-0400 [WARN]  ValidateProviderConfig from "provider[\"registry.terraform.io/hashicorp/azurerm\"]" changed the config value, but that value is unused
2024-08-09T16:25:24.463-0400 [DEBUG] provider.terraform-provider-azurerm_v3.108.0_x5: Configuring built-in cloud environment by name: "public": timestamp=2024-08-09T16:25:24.463-0400

--- a/docs/resources/compliance_security_profile_setting.md
+++ b/docs/resources/compliance_security_profile_setting.md
@@ -6,6 +6,8 @@ subcategory: "Settings"
 
 -> **Note** This resource could be only used with workspace-level provider!
 
+-> **Note** This setting can NOT be disabled once it is enabled.
+
 The `databricks_compliance_security_profile_workspace_setting` resource allows you to control whether to enable the 
 compliance security profile for the current workspace. Enabling it on a workspace is permanent. By default, it is 
 turned off. This setting can NOT be disabled once it is enabled.

--- a/settings/generic_setting.go
+++ b/settings/generic_setting.go
@@ -9,6 +9,7 @@ import (
 	"github.com/databricks/databricks-sdk-go"
 	"github.com/databricks/databricks-sdk-go/apierr"
 	"github.com/databricks/terraform-provider-databricks/common"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
 	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
 )
 
@@ -133,6 +134,10 @@ func (w workspaceSetting[T]) Update(ctx context.Context, c *databricks.Workspace
 	return w.updateFunc(ctx, c, t)
 }
 func (w workspaceSetting[T]) Delete(ctx context.Context, c *databricks.WorkspaceClient, etag string) (string, error) {
+	if w.deleteFunc == nil {
+		tflog.Warn(ctx, "The `delete` function isn't defined for this resource. Most probably it's not supported.")
+		return etag, nil
+	}
 	return w.deleteFunc(ctx, c, etag)
 }
 func (w workspaceSetting[T]) GetETag(t *T) string {
@@ -203,6 +208,10 @@ func (w accountSetting[T]) Update(ctx context.Context, c *databricks.AccountClie
 	return w.updateFunc(ctx, c, t)
 }
 func (w accountSetting[T]) Delete(ctx context.Context, c *databricks.AccountClient, etag string) (string, error) {
+	if w.deleteFunc == nil {
+		tflog.Warn(ctx, "The `delete` function isn't defined for this resource. Most probably it's not supported.")
+		return etag, nil
+	}
 	return w.deleteFunc(ctx, c, etag)
 }
 func (w accountSetting[T]) GetETag(t *T) string {

--- a/settings/resource_compliance_security_profile_setting.go
+++ b/settings/resource_compliance_security_profile_setting.go
@@ -6,6 +6,7 @@ import (
 
 	"github.com/databricks/databricks-sdk-go"
 	"github.com/databricks/databricks-sdk-go/service/settings"
+	"github.com/hashicorp/terraform-plugin-log/tflog"
 )
 
 // Enhanced Security Monitoring setting
@@ -31,5 +32,9 @@ var complianceSecurityProfileSetting = workspaceSetting[settings.ComplianceSecur
 			return "", err
 		}
 		return res.Etag, err
+	},
+	deleteFunc: func(ctx context.Context, w *databricks.WorkspaceClient, etag string) (string, error) {
+		tflog.Warn(ctx, "databricks_compliance_security_profile_workspace_setting couldn't be disabled!")
+		return etag, nil
 	},
 }

--- a/settings/resource_compliance_security_profile_setting_test.go
+++ b/settings/resource_compliance_security_profile_setting_test.go
@@ -256,3 +256,21 @@ func TestQueryUpdateComplianceSecurityProfileSettingWithConflict(t *testing.T) {
 	assert.Equal(t, true, res["is_enabled"])
 	assert.Equal(t, "HIPAA", res["compliance_standards"].([]interface{})[0])
 }
+
+func TestDeleteComplianceSecurityProfileSetting(t *testing.T) {
+	qa.ResourceFixture{
+		Resource: testComplianceSecurityProfileSetting,
+		Delete:   true,
+		HCL: `
+			compliance_security_profile_workspace {
+				is_enabled = true
+				compliance_standards = ["HIPAA", "PCI_DSS"]
+			}
+			etag = "etag1"
+		`,
+		ID: defaultSettingId,
+	}.ApplyAndExpectData(t, map[string]any{
+		etagAttrName: "etag1",
+		"id":         defaultSettingId,
+	})
+}

